initramfs/src/httpclient.rs

```diff
@@ -1,14 +1,24 @@
 use anyhow::{anyhow, Result};
 use reqwest::{Client, Response};
 
-pub async fn run_get_request(url: &str, token: Option<&str>) -> Result<Response> {
-    let res = match token {
+pub async fn run_get_request(
+    url: &str,
+    token: Option<&String>,
+    headers: Vec<(String, String)>,
```

Comment:
Why not use ``std::collections::HashMap`` with header names as key, if we ever need to act differently because of a specific header, it would come in handy to easily check if a key exists and to check its value.

Reply:
We can't: we need to set multiple values for a specific headers. For the "Accept" headers, we need to set multiple entries, which is not possible with an HashMap.